assets/wire/reducers.js

```diff
@@ -173,17 +174,31 @@ export default function wireReducer(state = initialState, action) {
             activeItem: action.item || null,
         };
 
-    case PREVIEW_ITEM:
+    case PREVIEW_ITEM: {
+        const readItems = Object.assign({}, state.readItems);
+
+        if (action.item) {
+            readItems[`${action.item._id}_${action.item.version}`] = true;
+        }
+
         return {
             ...state,
-            previewItem: action.item || null,
+            readItems,
+            previewItem: action.item ? action.item._id : null,
         };
+    }
+
+    case OPEN_ITEM:{
+        const readItems = Object.assign({}, state.readItems);
+        readItems[`${action.item._id}_${action.item.version}`] = true;
```

Comment:
this would add record for each item version, what about using `{item: version}` ?